File: Ruby/problem1.rb
load 'problem.rb'
require 'set'

class Problem1 < Problem

  def initialize(list, k, exp)
    @list = list
    @k = k
    @exp = exp
  end

  def Calculate()
    s1 = Set[]
    res = false
    for i in @list
      if s1.include?(i)
        res = true
      else
        s1.add(@k-i)
      end

    end

    return res

  end


end

File: PHP/Problem1.php
<?php
/**
 * Created by PhpStorm.
 * User: Animc
 * Date: 28-07-2018
 * Time: 12:57
 */



class Problem1 implements Problem
{


    public $list;
    public $k;
    public $exp;

    function __construct($list, $k, $exp)
    {
        $this->list = $list;
        $this->k = $k;
        $this->exp = $exp;
    }

    public function Calculate()
    {

    }
}